src/dwarf_parser.h

```diff
@@ -11,28 +13,37 @@
 
 namespace bpftrace {
 
+class BPFtrace;
+
 class Dwarf
 {
 public:
   virtual ~Dwarf();
 
-  static std::unique_ptr<Dwarf> GetFromBinary(const std::string &file_path);
+  static std::unique_ptr<Dwarf> GetFromBinary(BPFtrace *bpftrace,
+                                              const std::string &file_path);
 
   std::vector<std::string> get_function_params(
       const std::string &function) const;
 
+  ProbeArgs resolve_args(const std::string &function);
+
 private:
-  explicit Dwarf(const std::string &file_path);
+  explicit Dwarf(BPFtrace *bpftrace, const std::string &file_path);
```

Comment:
I don't see `bpftrace` ptr being used anywhere in this class. Is this intentional?

Reply:
Ah, good catch, it's not necessary here. I'll be needed once we support struct types, removing for now.